prime-router/src/main/kotlin/LookupTable.kt

```diff
@@ -57,6 +57,21 @@ class LookupTable(
         return table[rowNumber][colNumber]
     }
 
+    /**
+     * Filters a table down based on a predicate, and then invokes the standard lookupValue above
+     * on that filtered table below
+     */
+    fun lookupValue(
+        indexColumn: String,
+        indexValue: String,
+        lookupColumn: String,
+        filters: Map<String, String>,
+        ignoreCase: Boolean = true
+    ): String? {
+        val filteredTable = filter(filters, ignoreCase)
+        return filteredTable.lookupValue(indexColumn, indexValue, lookupColumn, ignoreCase)
+    }
+
```

Comment:
Adds a method that takes a map of filters and winnows down the LookupTable before then looking up values

Reply:
Suggest that you update the comment to include examples of how to use the filter. A Map<String, String> is a pretty generic type and doesn't communicate what the filter does.

Reply:
Would it better to think of this a second index or third index?

Reply:
It is potentially an index, but I didn't want to leave it limited to one or two options max. I wanted to add in the flexibility to get in a bunch of filters if we need to in the future.